File: src/de/saschahlusiak/freebloks/stats/StatisticsAdapter.java
package de.saschahlusiak.freebloks.stats;

import de.saschahlusiak.freebloks.R;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

public class StatisticsAdapter extends BaseAdapter {
	String labels[];
	String values1[];
	Context context;
	
	StatisticsAdapter(Context context, String[] labels, String[] values1) {
		this.context = context;
		this.labels = labels;
		this.values1 = values1;
	}
	
	@Override
	public int getCount() {
		return labels.length;
	}

	@Override
	public Object getItem(int position) {
		return null;
	}

	@Override
	public long getItemId(int position) {
		return position;
	}

	@Override
	public View getView(int position, View convertView, ViewGroup parent) {
		View v = LayoutInflater.from(context).inflate(R.layout.statistics_item, parent, false);
		((TextView)v.findViewById(android.R.id.text1)).setText(labels[position]);
		((TextView)v.findViewById(R.id.text2)).setText(values1[position]);
		return v;
	}

}

File: src/main/java/de/saschahlusiak/freebloks/statistics/StatisticsAdapter.kt
package de.saschahlusiak.freebloks.statistics

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import de.saschahlusiak.freebloks.R
import kotlinx.android.synthetic.main.statistics_item.view.*

class StatisticsAdapter(private val context: Context, private val labels: Array<String>, private val values1: Array<String?>) : BaseAdapter() {
    override fun getCount() = labels.size

    override fun getItem(position: Int) = null

    override fun isEnabled(position: Int) = (values1[position] != null)

    override fun areAllItemsEnabled() = true

    override fun getItemId(position: Int) = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val v = convertView ?: LayoutInflater.from(context).inflate(R.layout.statistics_item, parent, false)
        v.text1.apply {
            isEnabled = isEnabled(position)
            text = labels[position]
        }
        v.text2.apply {
            isEnabled = isEnabled(position)
            text = values1[position] ?: "--"
        }

        return v
    }
}